src/v/config/configuration.cc

```diff
@@ -1556,6 +1556,17 @@ configuration::configuration()
       "are downloaded.",
       {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
       false)
+  , cloud_storage_chunk_eviction_strategy(
+      *this,
+      "cloud_storage_chunk_eviction_strategy",
+      "Selects a strategy for evicting unused cache chunks.",
+      {.needs_restart = needs_restart::no,
+       .example = "predictive",
+       .visibility = visibility::tunable},
+      model::cloud_storage_chunk_eviction_strategy::predictive,
```

Comment:
We should discuss what the default will be here -- eager is very easy to understand, will work well for single sequential consumers, and will always enable the cache to be trimmed properly (i.e. avoid holding open file handles to files we try to trim).

The others are a bit harder to reason about: I'm mainly interested in the worst case scenarios, e.g. when using "capped" and the hydrated_chunks_per_segment_ratio is 0.7, would we always keep at least 5 chunks of each segment?   In that case our effective limit will really be `cloud_storage_max_materialized_segments_per_shard` multiplied by 5... which might be okay, but is kind of complicated.

Reply:
Hooking this in to `materialized_segments` is also an option, to apply a shard-global limit rather than trying to set per-segment limits and then have the actual limit be implicit based on the limit on materialized segments multiplied by the per-segment limits.

Reply:
> eager is very easy to understand, will work well for single sequential consumers, and will always enable the cache to be trimmed properly (i.e. avoid holding open file handles to files we try to trim).

I had actually started with the predictive version which was hardcoded with no strategy selection, but later added strategies and the other two models later because it might be easier for the end user to just simply  trim chunks which are not in use (although it might be nice to keep around the chunks which will soon be used, for example when multiple readers are operating with a small gap between them, so the first reader hydrates and the second one will just re-use the chunks).

> The others are a bit harder to reason about: I'm mainly interested in the worst case scenarios, e.g. when using "capped" and the hydrated_chunks_per_segment_ratio is 0.7, would we always keep at least 5 chunks of each segment? In that case our effective limit will really be `cloud_storage_max_materialized_segments_per_shard` multiplied by 5... which might be okay, but is kind of complicated.

Yes, the current implementation means that in the worst case, for each segment 0.7 times the total possible chunks (in that segment) can be kept around at a time. This is missing a global control point for the space usage right now where we can evict based on space usage across segments rather than per segment. 

> Hooking this in to materialized_segments is also an option, to apply a shard-global limit rather than trying to set per-segment limits and then have the actual limit be implicit based on the limit on materialized segments multiplied by the per-segment limits.

I had also thought about adding a shard wide control object to determine how many chunks to keep around/evict but wanted to address this in a later PR. I agree that `materialized_segments` could be a good place for this.